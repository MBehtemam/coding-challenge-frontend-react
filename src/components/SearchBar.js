import styled from 'styled-components';

const SearchBar = styled.section`
  background-color: #950740;
  height: 5vh;
`;

export default SearchBar;
