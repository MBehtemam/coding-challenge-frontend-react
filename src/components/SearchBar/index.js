import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import DatePicker from './components/DatePicker';
import QueryInput from './components/QueryInput';
import * as QueryActions from '../../logic/Actions/queryActions';
import * as OccurredBeforeActions from '../../logic/Actions/occuredBeforeActions';
import * as OccuredAfterActions from '../../logic/Actions/occuredAfterActions';
import Button from '../Button';

const Wrapper = styled.section`
  display: flex;
`;

const SearchBar = ({
  setQuery,
  query,
  occurredBefore,
  setOccurredBefore,
  setOccurredAfter,
  occurredAfter
}) => (
  <Wrapper>
    <QueryInput onBlur={e => setQuery(e.target.value)} defaultValue={query} />
    <DatePicker date={occurredBefore} title="From" onChagne={setOccurredBefore} />
    <DatePicker date={occurredAfter} title="To" onChagne={setOccurredAfter} />
    <Button>Clear</Button>
    <Button>Apply</Button>
  </Wrapper>
);
const mapStateToProps = state => ({
  query: state.query,
  occurredBefore: state.occurredBefore
});
const mapDispatchToProps = dispatch => ({
  setQuery: query => dispatch(QueryActions.setQuery(query)),
  setOccurredBefore: time => dispatch(OccurredBeforeActions.setOccurredBefore(time)),
  setOccurredAfter: time => dispatch(OccuredAfterActions.setOccurredAfter(time))
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SearchBar);

SearchBar.propTypes = {
  setQuery: PropTypes.func,
  query: PropTypes.string,
  occurredBefore: PropTypes.number,
  occurredAfter: PropTypes.number,
  setOccurredBefore: PropTypes.func,
  setOccurredAfter: PropTypes.func
};
SearchBar.defaultProps = {
  query: '',
  setQuery: QueryActions.setQuery,
  occurredBefore: 0,
  occurredAfter: 0,
  setOccurredBefore: OccurredBeforeActions.setOccurredBefore,
  setOccurredAfter: OccuredAfterActions.setOccurredAfter
};
