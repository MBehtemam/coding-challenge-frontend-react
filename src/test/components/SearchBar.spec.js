import React from 'react';
import SearchBar from '../../components/SearchBar';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('Test suits for SearchBar component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<SearchBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should has correct height', () => {
    const tree = renderer.create(<SearchBar />).toJSON();
    expect(tree).toHaveStyleRule('height', '5vh');
  });
});
