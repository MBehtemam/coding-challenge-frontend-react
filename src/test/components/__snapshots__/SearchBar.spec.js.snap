// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for SearchBar component should render correctly 1`] = `
.c0 {
  background-color: #950740;
  height: 5vh;
}

<section
  className="c0"
/>
`;
